test/unit/mdc-top-app-bar/mdc-top-app-bar.test.js

```diff
@@ -150,3 +160,45 @@ test('#adapter.deregisterNavigationIconInteractionHandler removes a handler from
   domEvents.emit(icon, 'click');
   td.verify(handler(td.matchers.anything()), {times: 0});
 });
+
+test('#adapter.registerScrollHandler adds a scroll handler to the window ' +
+  'element for a given event', () => {
+  const {component} = setupTest();
+  const handler = td.func('scrollHandler');
+  component.getDefaultFoundation().adapter_.registerScrollHandler(handler);
+
+  domEvents.emit(window, 'scroll');
+  try {
+    td.verify(handler(td.matchers.anything()));
+  } finally {
+    // Just to be safe
+    window.removeEventListener('scroll', handler);
+  }
+});
+
+test('#adapter.deregisterScrollHandler removes a scroll handler from the window ' +
+  'element for a given event', () => {
+  const {component} = setupTest();
+  const handler = td.func('scrollHandler');
+  window.addEventListener('scroll', handler);
+  component.getDefaultFoundation().adapter_.deregisterScrollHandler(handler);
+  domEvents.emit(window, 'scroll');
+  try {
+    td.verify(handler(td.matchers.anything()), {times: 0});
```

Comment:
Is this still going to properly fail the test case if we catch it?

Reply:
We're not catching the error, we're just using a finally to ensure the scroll handler gets unregistered from the window. I've checked that the test fails if the deregister fails.